TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Eel/FlowQueryOperations/ParentsUntilOperation.php

```diff
@@ -64,16 +64,14 @@ public function evaluate(FlowQuery $flowQuery, array $arguments)
             $siteNode = $contextNode->getContext()->getCurrentSiteNode();
             $parentNodes = $this->getParents($contextNode, $siteNode);
 
-            if (isset($arguments[0]) && !empty($arguments[0])) {
-                $untilQuery = new FlowQuery($parentNodes);
-                $untilQuery->pushOperation('filter', array($arguments[0]));
-
+            if (isset($arguments[0]) && !empty($arguments[0] && isset($parentNodes[0]))) {
+                $untilQuery = new FlowQuery(array($parentNodes[0]));
+                $untilQuery->pushOperation('closest', array($arguments[0]));
                 $until = $untilQuery->get();
             }
 
-            if (isset($until) && !empty($until)) {
-                $until = end($until);
```

Comment:
we also could rewrite `end($until)` to `$until[0]` in the faulty version (because that part caused the trouble in the first place). But i decided to take out the filter part completly (https://github.com/neos/neos-development-collection/pull/241/files#diff-da6eee0f189634f91c1b9d97abb888bcR69) and replace it by `closest`on the first parent. What is the better option?